coordinator/persistence-dse-next/src/main/java/io/stargate/db/cassandra/impl/EventListenerWrapper.java

```diff
@@ -0,0 +1,137 @@
+package io.stargate.db.cassandra.impl;
+
+import io.stargate.db.EventListener;
+import org.apache.cassandra.cql3.functions.UDAggregate;
+import org.apache.cassandra.cql3.functions.UDFunction;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.UserType;
+import org.apache.cassandra.schema.KeyspaceMetadata;
+import org.apache.cassandra.schema.SchemaChangeListener;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.schema.ViewMetadata;
+
+public class EventListenerWrapper implements SchemaChangeListener {
+  private final EventListener wrapped;
+
+  EventListenerWrapper(EventListener wrapped) {
+    this.wrapped = wrapped;
+  }
+
+  @Override
+  public void onCreateKeyspace(KeyspaceMetadata keyspace) {
```

Comment:
Should be change the listener to update cache using metadata directly? In the AbstractCassandraPersistence.java any change event we compute the schema for all keyspaces for any change. May not be done for this but we should see how this can be done.

Reply:
I think this was porting existing 4.0 logic as-is. Could be useful future optimization, although I am not sure what exactly is possible (it is not typically possible to re-calc hash code partially)

Reply:
Can someone make a ticket for this suggestion?